### DISCUSSION TITLE
AudioGridder plug-in UI shows left corner of screen when using Windows Remote Desktop

### DISCUSSION THREAD

**Discussion by beggehorn** [apohl79/audiogridder#327] [Q&A]
Just installed AudioGridder and am having my first experience with it. I'm currently accessing my ProTools rig remotely over a Windows Remote Desktop connection. When I load up AudioGridder VST (hosted in BlueCat Patchwork) and load a VST plug-in, the plug-in UI doesn't display but rather the upper left corner of my screen appears in the frame instead. I'm assuming this is being caused by accessing my desktop via RDP? Is there anything that can be done to address this?

Looking forwrad to having a chance to experiment more with a local session at the studio in the coming days.

I'm running Windows 10 Pro on a recently built i9-10980xe based machine, FWIW.

**Comment by CaBuffalo** [comment]
Hi, Can you screenshot the set up you have if possible?

Are you using your DAW, PT remotely and the Windows Pro 10 running on a 3rd
machine?

On Tue, Jan 26, 2021 at 7:21 PM beggehorn <notifications@github.com> wrote:

> Just installed AudioGridder and am having my first experience with it. I'm
> currently accessing my ProTools rig remotely over a Windows Remote Desktop
> connection. When I load up AudioGridder VST (hosted in BlueCat Patchwork)
> and load a VST plug-in, the plug-in UI doesn't display but rather the upper
> left corner of my screen appears in the frame instead. I'm assuming this is
> being caused by accessing my desktop via RDP? Is there anything that can be
> done to address this?
>
> Looking forwrad to having a chance to experiment more with a local session
> at the studio in the coming days.
>
> I'm running Windows 10 Pro on a recently built i9-10980xe based machine,
> FWIW.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/327>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6CTXUST2BDA7KMZIEDS36BFDANCNFSM4WUQS2LQ>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Reply by beggehorn** [reply]
I will do this later today.

To clarify a bit in the meantime, I am testing AudioGridder by running the server on the same machine as my DAW. The DAW computer is being accessed remotely via Remote Desktop (I work this way often from home when I can't get away to the studio). 

I'm primarily interested in seeing if AG can provide a solution to the unbalanced core loading in Protools caused by intensive native processing on my mix bus. I get CPU overload errors even though my CPU never shows more than 25% utilization.

**Reply by beggehorn** [reply]
Here's a screenshot of a simple ProTools 2020.12 session I created to illustrate the problem:

![Audiogridder mirrors upper left corner of screen](https://user-images.githubusercontent.com/78062974/106340716-9f54b800-624f-11eb-836e-0d1a92c95511.PNG)

It doesn't matter what VST3 plug-in (even amongst different 3ps) I try opening in AudioGridder, they all display the upper left corner of the screen in place of the loaded plug-ins UI. This display is "live" meaning if i open the File menu in ProTools, the AudioGridder plug-in window shows my movements and mirrors the action.

**Reply by beggehorn** [reply]
Oh wait!

I just happened to click on the plug-in name ("CAMEL") in AudioGridder and suddenly my screen looks like this:

![Audiogridder UI restored](https://user-images.githubusercontent.com/78062974/106341096-e7280f00-6250-11eb-8439-67922b038e50.PNG)

However if I try to move the AudioGridder Window by dragging the white header, the screen instantly reverts to my first screenshot. I can repeat this over and over. I'm also unable to interact with (either) Camel plug-in UI while it appears on my screen.

**Reply by apohl79** [reply]
AG server is intended to run on a separate machine or at least desktop. It always opens a plugin UI at the top left corner of the desktop and captures the screen to stream the UI to a remote machine running the AG plugin within your DAW and enable it to remote control the plugin.

On macOS it is possible to run multiple parallel user/login sessions on the same host using different user accounts. You might be able to do the same on windows, not sure. But if you can, you can run AG server on it's own desktop.

Other than that, your use case is not conveniently supported. What you could do is to turn off the screen capturing in the server and leave some space on your desktop to activate and control a plugin, which you still have to activate from the AG plugin.

**Reply by beggehorn** [reply]
OK, I think I understand.

I'm wondering if a simple feature request might make this situation more workable. Suppose in the case of having a multi-monitor setup, that a user could set a default display for the AG UI to appear? That combined with the ability to position the AG window freely (instead of being fixed in the upper right corner) would make my use case much more usable.

I do have a second PC at the studio but it is a much older machine and like I said, being able to access CPU cycles on my powerful i9 that ProTools is not currently utilizing would be my main goal.

**Reply by beggehorn** [reply]
Today, I updated to the latest release (nice!) and tried your suggestion of turning off the screen capturing. AG hosted plug-ins are working and I can interact with the UI. 

I really wish there were a way to dock the plug-in UI to the AG window (would likely need a redesign) and that the hosted plug-in wasn't forced to open in the upper right of the screen but rather opened in its last position (if relocated). 

I'm currently installing VST3 versions of some of the AAX plug-ins I use on my submasters/master aux tracks to test consumption, latency (incl. compensation) and how overall performance compares...

**Comment by Godlike-Productions** [comment]
I get the same behaviour (though haven't tried the release from 29 Jan yet)

**Reply by beggehorn** [reply]
My problems went away after updating to the 1,1,0 release and then setting Screen Capturing Mode to "Disabled" in AG Server Settings. I did both of these at the same time and didn't test after each individual change so I'm not sure which one fixed the issue of not being able to interact with the hosted plug-ins UI.

My setup of running the server and DAW on the same machine is working at the moment. Still testing and working out some performance kinks but will save those issues for their own threads...

**Comment by CaBuffalo** [comment]
Are you using RDP as well? If so, please refer to Andreas' reply above.

Try the latest release. See if it works.

On Sun, Jan 31, 2021 at 4:23 PM Godlike Productions <
notifications@github.com> wrote:

> I get the same behaviour (though haven't tried the release from 29 Jan yet)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/327#discussioncomment-326083>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6CIX3CHREPWYG4SN4DS4XX7NANCNFSM4WUQS2LQ>
> .
>


-- 
Kevin Ma
Supported by Gmail.

**Comment by CaBuffalo** [comment]
Good to hear. Thanks for the update.

On Mon, Feb 1, 2021 at 9:37 AM beggehorn <notifications@github.com> wrote:

> My problems went away after updating to the 1,1,0 release and then setting
> Screen Capturing Mode to "Disabled" in AG Server Settings. I did both of
> these at the same time and didn't test after each individual change so I'm
> not sure which one fixed the issue of not being able to interact with the
> hosted plug-ins UI.
>
> My setup of running the server and DAW on the same machine is working at
> the moment. Still testing and working out some performance kinks but will
> save those issues for their own threads...
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/apohl79/audiogridder/discussions/327#discussioncomment-328127>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARMAR6F7F3KP5PCR6V73E73S43RGVANCNFSM4WUQS2LQ>
> .
>


-- 
Kevin Ma
Supported by Gmail.
